tests/test_aea_exception_policy.py

```diff
@@ -70,15 +70,18 @@ def handler_func(*args, **kwargs):
         self.handler = handler_cls(name="handler1", skill_context=skill_context)
         self.behaviour = behaviour_cls(name="behaviour1", skill_context=skill_context)
 
-        test_skill = Skill(
-            SkillConfig(name="test_skill", author="fetchai"),
-            skill_context=skill_context,
-            handlers={"handler": self.handler},
-            behaviours={"behaviour": self.behaviour},
+        skill_class = Skill
+        skill_config = SkillConfig(name="test_skill", author="fetchai")
+        builder.add_component_instance(
+            skill_class,
+            configuration=skill_config,
+            args=[skill_config],
+            kwargs=dict(
+                skill_context=skill_context,
+                handlers={"handler": self.handler},
+                behaviours={"behaviour": self.behaviour},
```

Comment:
here's part of the problem: we can't deepcopy the components, since they contain the skill context (which in turn contains the agent context). And actually, they can contain arbitrary code that could be non-deep-copiable.

Maybe we should construct the skill component instances inside the builder as well.